model/distro/distro.go

```diff
@@ -37,6 +37,7 @@ type Distro struct {
 	PlannerSettings       PlannerSettings         `bson:"planner_settings" json:"planner_settings" mapstructure:"planner_settings"`
 	DispatcherSettings    DispatcherSettings      `bson:"dispatcher_settings" json:"dispatcher_settings" mapstructure:"dispatcher_settings"`
 	HostAllocatorSettings HostAllocatorSettings   `bson:"host_allocator_settings" json:"host_allocator_settings" mapstructure:"host_allocator_settings"`
+	ShallowClone          bool                    `bson:"shallow_clone" json:"shallow_clone" mapstructure:"shallow_clone"`
```

Comment:
do we want a global option to degrade shallow clone as a degraded mode flag?

Reply:
I don't think so. This is on the agent, so a global option is a little odd. It'd have to resolve in the distro that gets passed to the agent. I think we would turn it on a handful of popular distros, and then if it looks good after a couple days, turn it on everywhere.